tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedErrorMatcher.cs

```diff
@@ -0,0 +1,135 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format
+{
+    public class UnifiedErrorMatcher
+    {
+        public void AssertErrorsMatch(Exception actualException, BsonDocument expectedError)
+        {
+            expectedError.Elements.Should().NotBeEmpty();
```

Comment:
Replace with: 

    if (expectedError.ElementCount == 0)
    {
        throw new FormatException("expectedError document is empty.");
    }

`Should` should be reserved for things that mean the test failed.

`ElementCount == 0` is a bad test file, not a failed test.